### ISSUE TITLE
Naemon does not send email

### ISSUE THREAD

**Issue by mgadi** [naemon/naemon#76]
Hello, from Naemon 1.06 on Debian 8 X64, installed with offcial repository, I can't figure out but for some reason Naemon doesn't send me an alert on some of the services that I monitor, I was wondering if someone  can help me out a bit.

this is service 

```
define service {
service_description Root Partition
host_name XXXXXXXXXXX
use generic-service,service-pnp
check_command check_local_disk!20%!10%!/
contact_groups admins
}
```

here is my command 

```
define command {
command_name check_local_disk
command_line $USER1$/check_disk -w $ARG1$ -c $ARG2$ -p $ARG3$
}

```
```
define command {
  command_name                   notify-host-by-email
  command_line                   /usr/bin/printf "%b" "***** Naemon *****\n\nNotification Type: $NOTIFICATIONTYPE$\nHost: $HOSTNAME$\nState: $HOSTSTATE$\nAddress: $HOSTADDRESS$\nInfo: $HOSTOUTPUT$\n\nDate/Time: $LONGDATETIME$\n" \
                                  | /usr/bin/mail \
                                    -s "** $NOTIFICATIONTYPE$ Host Alert: $HOSTNAME$ is $HOSTSTATE$ **" $CONTACTEMAIL$
}

define command {
  command_name                   notify-service-by-email
  command_line                   /usr/bin/printf "%b" "***** Naemon *****\n\nNotification Type: $NOTIFICATIONTYPE$\n\nService: $SERVICEDESC$\nHost: $HOSTALIAS$\nAddress: $HOSTADDRESS$\nState: $SERVICESTATE$\n\nDate/Time: $LONGDATETIME$\n\nAdditional Info:\n\n$SERVICEOUTPUT$\n" \
                                  | /usr/bin/mail \
                                    -s "** $NOTIFICATIONTYPE$ Service Alert: $HOSTALIAS$/$SERVICEDESC$ is $SERVICESTATE$ **" $CONTACTEMAIL$
}
```

```
define contact {
contact_name                   XXXXXXX XXXXXX; Short name of user
alias Naemon                           XXXXXXXX; Full name of user
use generic-contact ; Inherit default values from generic-contact template (defined above)
email XXXXXXXX@XXXX.XX ; <<***** CHANGE THIS TO YOUR EMAIL ADDRESS ******
host_notification_commands notify-host-by-email
host_notification_options d,u
host_notification_period 24x7
host_notifications_enabled 1
service_notification_commands notify-service-by-email
service_notification_options c
service_notifications_enabled 1
}

```

```
define contactgroup {
    contactgroup_name        admins
    alias                                admins
    members                        admin 
    } 
```
sample in naemon.debug
```
[1506002239.652538] [032.0] [pid=1183] ** Service Notification Attempt ** Host: 'XXXXXXXXXXX', Service: 'Root Partition', Type: CUSTOM, Options: 3, Current State: 1, Last Notification: Thu Jan  1 01:00:00 1970
[1506002239.652551] [032.1] [pid=1183] This is a forced service notification, so we'll send it out.
[1506002239.652554] [032.0] [pid=1183] Notification viability test passed.
[1506002239.652557] [032.1] [pid=1183] Current notification number: 0 (changed)
[1506002239.652560] [032.2] [pid=1183] Creating list of contacts to be notified.
[1506002239.652563] [032.1] [pid=1183] Service notification will NOT be escalated.
[1506002239.652566] [032.1] [pid=1183] This notification will be BROADCAST to all (escalated and normal) contacts...
[1506002239.652568] [032.1] [pid=1183] Adding contacts from service escalation(s) to notification list.
[1506002239.652571] [032.1] [pid=1183] Adding normal contacts for service to notification list.
[1506002239.652573] [032.2] [pid=1183] Adding members of contact group 'admins' for service to notification list.
[1506002239.652576] [032.2] [pid=1183] ** Checking service notification viability for contact 'admin'...
[1506002239.652579] [032.1] [pid=1183] This is a forced service notification, so we'll send it out to this contact.
[1506002239.652581] [032.2] [pid=1183] Adding contact 'admin' to notification list.
[1506002239.652587] [032.2] [pid=1183] ** Notifying contact 'admin'
[1506002239.652595] [032.2] [pid=1183] Raw notification command: /usr/bin/printf "%b" "***** Naemon *****\n\nNotification Type: $NOTIFICATIONTYPE$\n\nService: $SERVICEDESC$\nHost: $HOSTALIAS$\nAddress: $HOSTADDRESS$\nState: $SERVICESTATE$\n\nDate/Time: $LONGDATETIME$\n\nAdditional Info:\n\n$SERVICEOUTPUT$\n" | /usr/bin/mail -s "** $NOTIFICATIONTYPE$ Service Alert: $HOSTALIAS$/$SERVICEDESC$ is $SERVICESTATE$ **" $CONTACTEMAIL$
[1506002239.652619] [032.2] [pid=1183] Processed notification command: /usr/bin/printf "%b" "***** Naemon *****\n\nNotification Type: CUSTOM\n\nService: Root Partition\nHost: XXXXXXXXXXX - NAGIOS - XX-XXXX\nAddress: XXXXXXXXXXX\nState: WARNING\n\nDate/Time: Thu Sept 21 15:57:19 CEST 2017\n\nAdditional Info:\n\nDISK WARNING - free space: / 1130 MB (14% inode=71%):\n" | /usr/bin/mail -s "** CUSTOM Service Alert: XXXXXXXXXXX- NAGIOS - XX-XXXX/Root Partition is WARNING **" XXXXXXXX@XXXX.XX
[1506002239.652651] [032.0] [pid=1183] 1 contacts were notified.
```
the timestamp 1506002239.652619 display this time : Thu, 21 Sep 2017 15:57:19 +02:00

sample postix log

![resultat-dans-postfix](https://user-images.githubusercontent.com/4088423/30749678-715e170e-9fb4-11e7-9c8b-cc7ee8662364.png)

the notification command work's in console

![commande-vers-postfix](https://user-images.githubusercontent.com/4088423/30749780-b6441666-9fb4-11e7-9b17-aabcd07eb4f9.png)

**Comment by pixnion** [comment]
IIRC...

naemon was improved in terms of actually terminating all processes related to a check execution (i.e. any children to a check process should be terminated). The same goes for all commands, including those related to notifications and event handlers.

mailx forks and spawns a child that handles the (possibly slow) communication with the local mail server, and then the parent process exits. The parent process exiting before this child process results in naemon thinking that the execution is finished and all remaining children should die => the child process terminates and the mail is never actually delivered.

TL;DR, just add the argument -Ssendwait to your /usr/bin/mail command line, and mailx shouldn't terminate prematurely. :)

**Comment by mgadi** [comment]
I have set this variable option in /etc/mail.rc, it does'nt work either. 
here is my command without the variable option set in /etc/mail.rc

```
define command {
  command_name                   notify-host-by-email
  command_line                   /usr/bin/printf "%b" "***** Naemon *****\n\nNotification Type: $NOTIFICATIONTYPE$\nHost: $HOSTNAME$\nState: $HOSTSTATE$\nAddress: $HOSTADDRESS$\nInfo: $HOSTOUTPUT$\n\nDate/Time: $LONGDATETIME$\n" \
                                  | /usr/bin/mail \
                                    -Ssendwait \
                                    -s "** $NOTIFICATIONTYPE$ Host Alert: $HOSTNAME$ is $HOSTSTATE$ **" $CONTACTEMAIL$
}
```
```
define command {
  command_name                   notify-service-by-email
  command_line                   /usr/bin/printf "%b" "***** Naemon *****\n\nNotification Type: $NOTIFICATIONTYPE$\n\nService: $SERVICEDESC$\nHost: $HOSTALIAS$\nAddress: $HOSTADDRESS$\nState: $SERVICESTATE$\n\nDate/Time: $LONGDATETIME$\n\nAdditional Info:\n\n$SERVICEOUTPUT$\n" \
                                  | /usr/bin/mail \
                                    -Ssendwait \
                                    -s "** $NOTIFICATIONTYPE$ Service Alert: $HOSTALIAS$/$SERVICEDESC$ is $SERVICESTATE$ **" $CONTACTEMAIL$
}
```

**Comment by mgadi** [comment]
It work's ! Our configuration have several template. Some values in the directives used was overwritten by different hosts definition so i decided to define template for branches depending on the pririoty or type of service.

**Comment by jacobbaungard** [comment]
Closing this as the comment above seems to indicate the this issue is resolved.
